### PR TITLE
BIIIIGGGG CHANGES OWO

### PR DESCRIPTION
**Pull request by Vendicated**
- Parse path ourselves and resolve files accordingly
- Properly parse args:
  - Expand variables via $
  - Allow expanding with ${} so something like `owo=${USER}iscute` will expand to `owo=veniscute`
  - Whitespace will not break arg inside single or double quotes
  - variables will not be expanded inside single quotes
  - You can escape any of the above via \\$USER (Needs double escape; this might be a limitation with nodejs, not sure)
- You can now set session variables via `key=value` or command invoke variables via `key=value command`
- New builtins:
   - export
   - which
- pnpm lock should not be gitignored
- Flag parser removed for now, but I already have a working one here https://github.com/Vendicated/EmoteBot/blob/main/src/commands/CommandArguments.ts#L35-L49 so we can add that back

### CODE REVIEW

File: src/commands/export.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { InbuiltCommand } from "../classes";
+import { ExitCodes } from "../constants";
+
+export default class Export extends InbuiltCommand {
+  declare usage: "KEY=VALUE";
```
Comment (nearlySplat) [inline src/commands/export.ts:5]: Shouldn't this be `=`, since you're just declaring the value?

File: src/handleCommand.ts [hunk 2]
```diff
@@ -1,41 +1,40 @@
...
+          .join(" ")} ${commandName} ${args.map(arg => `"${arg}"`).join(" ")}`,
+        { shell: true }
+      );
       child.stdout.on("data", (d) => process.stdout.write(d.toString()));
       child.on("exit", (code, signal) => r({ out: "", code: code || signal }));
     });
   }
-  static isSyntaxError(str: string) {
-    return /(^&$|^function\s*$|^fn\s*$)/.test(str);
-  }
```
Comment (nearlySplat) [inline src/handleCommand.ts:39]: I don't really know why I put these there even though I didn't even have a `fn` command,  might add a TODO comment there

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { ExitCodes } from "./constants";
+
+export type PotentialPromise<T> = Promise<T> | T;
```
Comment (nearlySplat) [inline src/types.ts:3]: TypeScript has a `PromiseLike` type for this IIRC.
```suggestion
export type PotentialPromise<T> = PromiseLike<T>;
```

**Review by nearlySplat — COMMENTED** [review]
Other than those nitpicks, this looks ready to commit!

File: src/handleCommand.ts (hunk 2 above)
Comment (Vendicated) [reply src/handleCommand.ts:39]: yeah I was kinda confused by this so I just removed it lmao. what would a fn command be like? we could just allow Javascript to be entered and eval it (in a different context so it doesn't affect splatsh internals)

**Pull request merged**